lib/util/http/github.ts

```diff
@@ -315,9 +320,20 @@ export class GithubHttp extends Http<GithubHttpOptions> {
           if (!process.env.RENOVATE_PAGINATE_ALL && opts.paginate !== 'all') {
             lastPage = Math.min(pageLimit, lastPage);
           }
+          const baseUrl = opts.baseUrl;
+          const parsedUrl = new URL(linkHeader.next!.url, baseUrl);
+          const rebasePagination =
+            !!baseUrl &&
+            !!process.env.RENOVATE_X_REBASE_PAGINATION_LINKS &&
+            // Preserve github.com URLs for use cases like release notes
+            parsedUrl.origin !== 'https://api.github.com';
+          const firstPageUrl = rebasePagination
+            ? replaceUrlBase(parsedUrl, baseUrl)
+            : parsedUrl;
           const queue = [...range(2, lastPage)].map(
             (pageNumber) => (): Promise<HttpResponse<T>> => {
-              const nextUrl = new URL(linkHeader.next!.url, opts.baseUrl);
+              // copy before modifying searchParams
+              const nextUrl = new URL(firstPageUrl);
```

Comment:
use `linkHeader.next.url` here if rebase is false, not first page url.

Reply:
Do you mean `new URL(linkHeader.next!.url, baseUrl)`?  That's what `firstPageUrl` is when `rebasePagination` is false.

Reply:
right, sorry. my fault.